pkg/autodetect/main.go

```diff
@@ -149,6 +151,14 @@ func (b *Background) autoDetectCapabilities() {
 		b.detectKafka(apiList)
 	}
 
+	watchNamespace, err := k8sutil.GetWatchNamespace()
```

Comment:
Please, create a new func to hold this logic. Also, please check what's the error you get back. Just receiving an error might not be sufficient to make the right decision: on the first run of this func, this might correctly detect that the scope is `cluster`, but a future unrelated failure (network failure when talking to some backend API server) will change the scope to `namespace`.